src/dhcpcd-qt/dhcpcd-wi.cpp

```diff
@@ -295,13 +295,21 @@ void DhcpcdWi::ping()
 void DhcpcdWi::connectSsid(DHCPCD_WI_SCAN *scan)
 {
 	DHCPCD_WI_SCAN s;
+	DHCPCD_IF *i;
 	int err;
 
 	/* Take a copy of scan incase it's destroyed by a scan update */
 	memcpy(&s, scan, sizeof(s));
 	s.next = NULL;
 
-	if (s.flags & WSF_PSK) {
+	i = dhcpcd_wpa_if(wpa);
```

Comment:
`i` needs to be checked for NULL here.

Reply:
Should I define the error value in another header entry?

Reply:
Use `DHCPCD_WPA_ERR`